extensions/mssql/src/ui/scriptableDialogBase.ts

```diff
@@ -95,7 +102,9 @@ export abstract class ScriptableDialogBase<OptionsType extends ScriptableDialogO
 			}
 			let message: string;
 			const script = await this.generateScript();
-			if (script) {
+			if (this.opensEditorSeparately) {
```

Comment:
The existing backup & restore STS methods do their operations in the background, so we don't get a script back from the method call. This change is so we can return immediately without waiting on results.